### PR TITLE
Nytwidget

### PR DESCRIPTION
**Pull request by david-novak**
Nisem prepričan zakaj je en commit podvojen ... Ampak zdaj bi morala biti zadnja verzija na mojem repozitoriju.

### CODE REVIEW

**Comment by kernc** [comment]
You can rebase this branch to even out the oddities:
https://help.github.com/articles/about-git-rebase/
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request
https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

File: orangecontrib/text/corpus.py [hunk 1]
```diff
@@ -28,16 +28,16 @@ def __init__(self, documents, metadata=None):
             for key in md.keys():
                 metadata2.setdefault(key, set()).add(md[key])
 
-        meta_vars = []
+        self.meta_vars = []
```
Comment (nikicc) [inline orangecontrib/text/corpus.py]: Why do you need metas attached to self? They shoud be accessed from the Corpus, if need later. Probably the same shuld be fixed for documents.

File: orangecontrib/text/corpus.py [hunk 2]
```diff
@@ -78,6 +78,37 @@ def from_file(cls, filename):
                 metadata.append({header[i]: fields[i] for i in range(len(fields)) if i != text_ind})
         return cls(documents, metadata)
 
+    def extend_corpus(self, documents, metadata):
+        metadata2 = dict()
```
Comment (nikicc) [inline orangecontrib/text/corpus.py]: This is a copy of the above code. It should be put into a method to eliminate repetition.

File: orangecontrib/text/corpus.py [hunk 3]
```diff
@@ -78,6 +78,37 @@ def from_file(cls, filename):
...
+            for j, var in enumerate(meta_vars):
+                if var.name in md:
+                    metas[i][j] = var.to_val(md[var.name])
+
+        text = np.array(documents).reshape(len(documents), 1)
+        metas = np.array(metas, dtype=object)
+
+        additional_metas = np.hstack((text, metas))
+        # Update by adding the new metas and documents.
+        self.metas = np.vstack((self.metas, additional_metas))
```
Comment (nikicc) [inline orangecontrib/text/corpus.py]: Doesnt this also add metas that are already present in Corpus?

File: orangecontrib/text/corpus.py [hunk 4]
```diff
@@ -78,6 +78,37 @@ def from_file(cls, filename):
...
+        self.documents += documents
+
+        # Update the Domain.
+        self.meta_vars += meta_vars
```
Comment (nikicc) [inline orangecontrib/text/corpus.py]: Doesn't this also add metas that are already present in Corpus?

File: orangecontrib/text/nyt.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+        :param record: The JSON representation of the query's results.
+        :return: A list of articles parsed into documents and a list of the
+            corresponding metadata, joined in a tuple.
+        """
+        documents = []
+        metadata = []
+        for doc in record["response"]["docs"]:
+            documents.append(self._walk_json(doc, ""))
+            metadata.append(dict(category="article"))
```
Comment (nikicc) [inline orangecontrib/text/nyt.py]: Why do all the documents have category "article"?

File: orangecontrib/text/nyt.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        return documents, metadata
+
+    def _walk_json(self, json_document, string_document=""):
```
Comment (nikicc) [inline orangecontrib/text/nyt.py]: Is `string_document ` really needed if you always call it with an empty string?

File: orangecontrib/text/nyt.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                string_document += " " + item
+        return string_document
+
+    @staticmethod
+    def validate_year(year):
+        if int(year) <= 1851:
+            return "1851"
+        if int(year) >= date.today().year:
+            return str(date.today().year)
+        return year
```
Comment (nikicc) [inline orangecontrib/text/nyt.py]: Enter missing.

**Comment by nikicc** [comment]
File `ownyt.py` for sure needs refactoring, all the logic should be moved out.

**Pull request merged**